lib/bloc/fee_options/fee_options_bloc.dart

```diff
@@ -17,54 +17,76 @@ class FeeOptionsBloc extends Cubit<FeeOptionsState> {
     try {
       recommendedFees = await _breezSDK.recommendedFees();
       _log.info(
         "fetchFeeOptions recommendedFees:\nfastestFee: ${recommendedFees.fastestFee},"
         "\nhalfHourFee: ${recommendedFees.halfHourFee},\nhourFee: ${recommendedFees.hourFee}.",
       );
-      return await _constructFeeOptionList(toAddress, recommendedFees);
+      return await _constructFeeOptionList(
+        swapAddress: swapAddress,
+        toAddress: toAddress,
+        recommendedFees: recommendedFees,
+      );
     } catch (e) {
       _log.severe("fetchFeeOptions error", e);
       emit(FeeOptionsState(error: extractExceptionMessage(e, getSystemAppLocalizations())));
       rethrow;
     }
   }
 
-  Future<List<FeeOption>> _constructFeeOptionList(
-    String toAddress,
-    RecommendedFees recommendedFees,
-  ) async {
+  Future<List<FeeOption>> _constructFeeOptionList({
+    required String toAddress,
+    required RecommendedFees recommendedFees,
+    String? swapAddress,
+  }) async {
     final List<FeeOption> feeOptions = [
       FeeOption(
         processingSpeed: ProcessingSpeed.economy,
         waitingTime: const Duration(minutes: 60),
-        fee: await _calculateTransactionFee(toAddress, recommendedFees.hourFee),
+        fee: await _calculateTransactionFee(
+            toAddress: toAddress, satPerVbyte: recommendedFees.hourFee, swapAddress: swapAddress),
         feeVByte: recommendedFees.hourFee,
       ),
       FeeOption(
         processingSpeed: ProcessingSpeed.regular,
         waitingTime: const Duration(minutes: 30),
-        fee: await _calculateTransactionFee(toAddress, recommendedFees.halfHourFee),
+        fee: await _calculateTransactionFee(
+            toAddress: toAddress, satPerVbyte: recommendedFees.halfHourFee, swapAddress: swapAddress),
         feeVByte: recommendedFees.halfHourFee,
       ),
       FeeOption(
         processingSpeed: ProcessingSpeed.priority,
         waitingTime: const Duration(minutes: 10),
-        fee: await _calculateTransactionFee(toAddress, recommendedFees.fastestFee),
+        fee: await _calculateTransactionFee(
+            toAddress: toAddress, satPerVbyte: recommendedFees.fastestFee, swapAddress: swapAddress),
         feeVByte: recommendedFees.fastestFee,
       ),
     ];
     emit(state.copyWith(feeOptions: feeOptions));
     return feeOptions;
   }
 
-  Future<int> _calculateTransactionFee(String toAddress, int satsPerVbyte) async {
+  Future<int> _calculateTransactionFee({
+    required toAddress,
+    required int satPerVbyte,
+    String? swapAddress,
+  }) async {
+    // When the swap address is present we proceed with calculating a prepare refund request.
+    if (swapAddress != null) {
+      final req = PrepareRefundRequest(
+        swapAddress: swapAddress,
+        toAddress: toAddress,
+        satPerVbyte: satPerVbyte,
+      );
+      final response = await _breezSDK.prepareRefund(req: req);
+      return response.refundTxFeeSat;
```

Comment:
Missing error handling, see `prepareRefund` on `RefundBloc`. Same goes for `prepareSweep` logic.

Internal logic of ` _calculateTransactionFee` has to be more distinctive and less prone to errors:
`return (swapAddress != null) ? prepareRefund(...) : prepareSweep(...);`

Reply:
Done